drivers/net/bnxt/bnxt_hwrm.c

```diff
@@ -951,6 +955,54 @@ int bnxt_hwrm_vnic_rss_cfg(struct bnxt *bp,
 	return rc;
 }
 
+#define HW_MAX_VNICS 170
+
+int bnxt_hwrm_func_vf_stall(struct bnxt *bp, uint16_t vf, uint8_t on)
+{
+	struct hwrm_func_vf_vnic_ids_query_input req = {0};
+	struct hwrm_func_vf_vnic_ids_query_output *resp = bp->hwrm_cmd_resp_addr;
+	struct bnxt_vnic_info vnic;
+	int rc;
+	uint32_t i, num_vnic_ids;
+	uint16_t *vnic_ids;
+
+	/* First query all VNIC ids */
+
+	vnic_ids = rte_malloc("bnxt_hwrm_vf_vnic_ids_query", HW_MAX_VNICS * sizeof(*vnic_ids),
+			RTE_CACHE_LINE_SIZE);
+	if (vnic_ids == NULL) {
+		rc = -ENOMEM;
+		return rc;
+	}
+
+	HWRM_PREP(req, FUNC_VF_VNIC_IDS_QUERY, -1, resp_vf_vnic_ids);
+
+	req.vf_id = rte_cpu_to_le_16(bp->pf.first_vf_id + vf);
+	req.max_vnic_id_cnt = rte_cpu_to_le_32(HW_MAX_VNICS);
+	req.vnic_id_tbl_addr = rte_malloc_virt2phy(vnic_ids);
+
+	rc = bnxt_hwrm_send_message(bp, &req, sizeof(req));
+	HWRM_CHECK_RESULT;
+
+	num_vnic_ids = resp->vnic_id_cnt;
+
+	/* Retrieve VNIC, update bd_stall then update */
+
+	for (i = 0; i < num_vnic_ids; i++) {
+		memset(&vnic, 0, sizeof(struct bnxt_vnic_info));
+		vnic.fw_vnic_id = vnic_ids[i];
+		vnic.bd_stall = on;
+		//TODO bnxt_hwrm_vnic_qcfg(&vnic);
+		//rc = bnxt_hwrm_vnic_cfg(bp, &vnic);
```

Comment:
I'll complete this if the direction is correct, in other words, introduce hwrm_vnic_qcfg message, read vnic info and update by reusing bnxt_hwrm_vnic_cfg().